src/functions/secrets.ts

```diff
@@ -137,17 +166,28 @@ export async function pruneSecrets(
   // Collect all Firebase managed secret versions
   const haveSecrets = await listSecrets(projectId, `labels.${FIREBASE_MANGED}=true`);
   for (const secret of haveSecrets) {
-    const versions = await listSecretVersions(projectId, secret.name, `state: ENABLED`);
+    const versions = await listSecretVersions(projectId, secret.name, `NOT state: DESTROYED`);
     for (const version of versions) {
       prunedSecrets.add(pruneKey(secret.name, version.versionId));
     }
   }
 
   // Prune all project-scoped secrets in use.
-  const sevs = of(endpoints).filter(
-    (sev) => sev.projectId === projectId || sev.projectId === projectNumber
-  );
-  for (const sev of sevs) {
+  const secrets: Required<backend.SecretEnvVar>[] = [];
+  for (const secret of of(endpoints)) {
+    if (!secret.version) {
+      // All bets are off if secret version isn't available in the endpoint definition.
+      // This should never happen for GCFv1 instances.
+      throw new FirebaseError(`Secret ${secret.secret} version is unexpectedly empty.`);
+    }
+    if (secret.projectId === projectId || secret.projectId === projectNumber) {
+      if (secret.version) {
```

Comment:
Do we really need this check since we have L178?

Reply:
I added this check to make Typescript compiler happy haha. If I remove this check,  `secrets.push({ ...secret, version: secret.version });` `tsc` emits a warning. I'll add a comment to explain why this check exists.

Reply:
oh nvm - you are right. This check is redundant - Typescript is able to figure out that this won't be null.